tests/es-basic.sh

```diff
@@ -16,21 +17,60 @@ add_conf '
 template(name="tpl" type="string"
 	 string="{\"msgnum\":\"%msg:F,58:2%\"}")
 
+module(load="../plugins/impstats/.libs/impstats" interval="1"
+	   log.file="'"$RSYSLOG_DYNNAME.spool"'/omelasticsearch-stats.log" log.syslog="off" format="cee")
 module(load="../plugins/omelasticsearch/.libs/omelasticsearch")
 
 if $msg contains "msgnum:" then
 	action(type="omelasticsearch"
 	       server="127.0.0.1"
 	       serverport="'$ES_PORT'"
 	       template="tpl"
-	       searchIndex="rsyslog_testbench")
+	       searchIndex="rsyslog_testbench"
+	       rebindinterval="'$REBIND_INTERVAL'")
 '
 startup
 injectmsg  0 $NUMMESSAGES
+wait_queueempty
+# now, wait for $NUMMESSAGES to be reported in stats file e.g.
+# Fri Mar 29 02:29:04 2019: @cee: { "name": "omelasticsearch", "origin": "omelasticsearch", "submitted": 1000, ...
+wait_content '"name": "omelasticsearch".*"submitted": '$NUMMESSAGES $RSYSLOG_DYNNAME.spool/omelasticsearch-stats.log
```

Comment:
Wait for the stats file to report that omelasticsearch has processed all of the records

Reply:
line 37 should solve the issue - If you agree, I'll add a commit which combines this via the new method used for this kind - it's a bit faster.

actually
```
export QUEUE_EMPTY_CHECK_FUNC=es_shutdown_empty_check
```
is already part of it. But it's easier to show in code than english text ;-)

Reply:
> If you agree, I'll add a commit which combines this via the new method used for this kind

yes, please do

Reply:
will do soon (finishing up things before switching)